File: app/src/main/java/com/simplemobiletools/musicplayer/receivers/IncomingCallReceiver.java
package com.simplemobiletools.musicplayer.receivers;

import android.content.Context;
import android.telephony.PhoneStateListener;
import android.telephony.TelephonyManager;

import com.simplemobiletools.musicplayer.Constants;
import com.simplemobiletools.musicplayer.Utils;

public class IncomingCallReceiver extends PhoneStateListener {
    private static Context mContext;

    public IncomingCallReceiver(Context context) {
        mContext = context;
    }

    @Override
    public void onCallStateChanged(int state, String incomingNumber) {
        super.onCallStateChanged(state, incomingNumber);

        if (state == TelephonyManager.CALL_STATE_RINGING) {
            Utils.sendIntent(mContext, Constants.CALL_START);
        } else if (state == TelephonyManager.CALL_STATE_IDLE) {
            Utils.sendIntent(mContext, Constants.CALL_STOP);
        }
    }
}

File: app/src/main/kotlin/com/simplemobiletools/musicplayer/receivers/IncomingCallReceiver.kt
package com.simplemobiletools.musicplayer.receivers

import android.content.Context
import android.telephony.PhoneStateListener
import android.telephony.TelephonyManager
import com.simplemobiletools.musicplayer.extensions.sendIntent
import com.simplemobiletools.musicplayer.helpers.CALL_START
import com.simplemobiletools.musicplayer.helpers.CALL_STOP

class IncomingCallReceiver(val context: Context) : PhoneStateListener() {

    override fun onCallStateChanged(state: Int, incomingNumber: String) {
        super.onCallStateChanged(state, incomingNumber)

        if (state == TelephonyManager.CALL_STATE_RINGING) {
            context.sendIntent(CALL_START)
        } else if (state == TelephonyManager.CALL_STATE_IDLE) {
            context.sendIntent(CALL_STOP)
        }
    }
}
